api/all/src/main/java/io/opentelemetry/api/logs/LogRecord.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.logs;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Used to construct and emit logs from a {@link Logger}.
+ *
+ * <p>Obtain a {@link LogRecordBuilder} via {@link Logger#eventBuilder(String)}, add properties using the
+ * setters, and emit the event by calling {@link #emit()}.
+ */
+public interface LogRecord {
```

Comment:
Do we want to make LogRecord part of public API?
Do we want to have both LogRecord with setters and LogRecordBuilder?